packages/fast-components-react-base/src/auto-suggest/auto-suggest.tsx

```diff
@@ -214,17 +255,33 @@ class AutoSuggest extends Foundation<
         state: AutoSuggestState,
         onChange: (event: React.ChangeEvent) => void,
         onClick: (event: React.MouseEvent) => void,
-        onKeyDown: (event: React.KeyboardEvent) => void
+        onKeyDown: (event: React.KeyboardEvent) => void,
+        onFocus: (event: React.FocusEvent) => void
     ): React.ReactNode => {
         const listboxId: string = state.isMenuOpen ? props.listboxId : null;
         const activedescendantId: string =
             state.focusedItem !== null ? state.focusedItem.id : null;
+        let menuTriggerProps: Partial<AutoSuggestUnhandledProps>;
+        switch (this.props.menuTrigger) {
+            case MenuTriggers.onFocus:
+                menuTriggerProps = {
+                    onFocus,
+                    onKeyDown,
+                };
+                break;
+
+            default:
+                menuTriggerProps = {
+                    onClick,
+                    onKeyDown,
```

Comment:
You could keep the logic in the render function to a minimum by hooking up all events here and deciding whether or not to act on them when you catch them.  The less logic we have in these customizable functions the better as any change we make needs to be picked up by folks who have rolled their own.